src/test/ui/regions/regions-assoc-type-in-supertrait-outlives-container.rs

```diff
@@ -49,7 +47,8 @@ fn with_assoc<'a,'b>() {
     // outlive 'a. In this case, that means TheType<'b>::TheAssocType,
     // which is &'b (), must outlive 'a.
 
-    let _: &'a WithAssoc<TheType<'b>> = loop { }; //~ ERROR reference has a longer lifetime
+    let _x: &'a WithAssoc<TheType<'b>> = loop { };
+    //~^ ERROR reference has a longer lifetime
```

Comment:
Can we ope an issue regarding enforcing WF conditions in dead code and put a FIXME here like this:

---

FIXME(#XXX) -- NLL doesn't enforce WF conditions (or any type annotations) in dead code.